Flexibleidealist/native single project

flexibleidealist opened the pull request:
- EditProject screen incorporates AboutProject component
- AboutProject deleted
- Screens that redirect to CreateProject screen will have to redirect to EditProject and pass in a `createNewProject` prop in the `route.params` with value `true` in order to set project properties to their default values
- latest changes address a bug that prevented a tool from being removed from the tools list

flexibleidealist committed:
sets up basic structure for SingleProject screen in native

flexibleidealist committed:
merged changes from otehr branch

flexibleidealist committed:
building out EditProject screen and preparing to connect it to SingleProject screen

flexibleidealist committed:
working on AboutProject component

flexibleidealist committed:
adds conditional rendering of an owner View to SingleProject screen to show Edit Project button and list of applicants; continues updating AboutProject component

flexibleidealist committed:
adds latest changes from native-development branch"

flexibleidealist committed:
adds interested users to SingleProject view for owners

flexibleidealist committed:
updated AboutProject to handle changing number of designers and engineers

flexibleidealist committed:
continues working on EditProject view; struggles with Picker

yeezick committed:
merging with my working repo which includes all other working screens
Merge branch 'yeezick/native-roulette' into flexibleidealist/native-single-project

yeezick committed:
- broke down AboutProject into 3 respective components: EditTools, EditTeamCount, EditTimeCommitment
- restructure EditProject to render a project based on params-projectID
- component flow is currently not breaking but also not done

yeezick committed:
- resolve state-related errors in editing project info
- remove console.logs from native application
this commit leaves EditProject in a working state but not functional, still lacks successful connection to backend server on update requests

yeezick committed:
- wrap up functionality for AboutProject meaning that users can successfully update or create a new project
- add navigation-based handling for SingleProject

flexibleidealist committed:
refactoring to eliminate AboutProject and consolidate with EditProject

flexibleidealist committed:
clean-up of EditProject screen in progress

flexibleidealist committed:
Continues EditProject cleanup; deletes AboutProject component; new bug on attempt to remove tool: error msg = Attempted to assign to readonly property.

flexibleidealist committed:
fixes bug that prevented removing a tool from the tools list in EditProject screen

flexibleidealist committed:
Prettier formatting attempt

yeezick commented on native/services/api/apiConfig.js:
```diff
@@ -1,9 +1,11 @@
 import axios from "axios";
+
```
remove whitespace here and line 8

yeezick commented on backend/controllers/projects.js:
```diff
@@ -19,7 +19,7 @@ import Tool from '../models/tool.js';
 export const getAllProjects = async (req, res) => {
   try {
-    const projects = await Project.find().populate([{path: 'team_members', model: User}]); // .populate({path: "interested_applicants",model: User,  });
+    const projects = await Project.find().populate([{path: 'team_members', model: User}, {path: 'tools', model: Tool}, {path: 'interested_applicants', model: User}]); // .populate({path: "interested_applicants",model: User,  });
```
create a ticket on jira, assign it to "back-end refactoring" epic (create it if it doesnt exist), and fill in title/description around the discussion of optimizing endpoints so that the application only holds information it needs. this will help improve memory-efficiency in the long run

yeezick commented:
looks good so far, just a few points to tackle:
- create project is not integrated (understood though since drawer-navigation complicates things)
- `EditProject` has a number of states can exist in the sub-components and not have to be passed down
i can take a look at these between today and monday, including taking a look at seeing if there's an opportunity to shorten the file by separating components by file

yeezick committed:
- relocate some states in EditPRoject to live in their components
- test out conditional variable definitions, subject to change/discussion
- replace functions updating objects in state to use the custom handleTextChange function which should be renamed as it only works with objects
- close out conversations from PR
- add a function declaration statement to handleTextChange to explain the function in order to use it properly as I confused myself with it

yeezick committed:
- comment out and remove some dead code from app.js
- create a folder for EditProject and its subcomponents --subject to change--
- attempt to clean up conditionals by defining conditional values at once instead of individually --subject to change--
- clean up prop passing in EditProject to avoid passing down unnecessary props --incomplete, have a good feeling the designer/engineer count may also be relocated--
- begin to work on tool-rendering bug where tools part of a project may be added to the project twice --this is due to projectTools and allTools being different objects in memory, therefore failing .includes method--
- subtle renaming in EditProject/helpers for semanticism

yeezick committed:
- add createNewProject to EditProject dependency
- update tool picking component to only show tools that should be available to the user
- tool picking component also has a default option and now avoids duplication of tools
